src/Util/CallableArgumentsExtractor.php

```diff
@@ -163,7 +166,17 @@ public function extractArguments(Node $arguments): array
 
     private function normalizeName(string $name): string
     {
-        return strtolower(preg_replace(['/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'], ['\\1_\\2', '\\1_\\2'], $name));
+        return strtolower(str_replace('_', '', $name));
+    }
+
+    // logic from Symfony\Component\String\AbstractUnicodeString::snake()
+    private function toSnakeCase(string $str): string
+    {
+        $camel = str_replace(' ', '', preg_replace_callback('/\b.(?!\p{Lu})/u', static function ($m) use (&$i) {
+            return 1 === ++$i ? ('İ' === $m[0] ? 'i̇' : mb_strtolower($m[0], 'UTF-8')) : mb_convert_case($m[0], \MB_CASE_TITLE, 'UTF-8');
+        }, preg_replace('/[^\pL0-9]++/u', ' ', $str)));
+
+        return mb_strtolower(preg_replace(['/(\p{Lu}+)(\p{Lu}\p{Ll})/u', '/([\p{Ll}0-9])(\p{Lu})/u'], '\1_\2', $camel), 'UTF-8');
```

Comment:
since normalizeName isn't unicode-aware, I suggest simplifying here:
```suggestion
    private function toSnakeCase(string $str): string
    {
        return strtolower(preg_replace(['/([A-Z]+)([A-Z][a-z])/', '/([a-z0-9])([A-Z])/'], '\1_\2', $str));
```

Reply:
if we want to keep unicode casing, we should likely do it only when the charset is utf-?8?